fix(docs): clarify CronWorkflow `startingDeadlineSeconds`. Fixes #12971 (#13280)
argoproj/argo-workflows: docs/cron-workflows.md

> 

Reviewer: I can't comment on unchanged lines but the description of `activeDeadlineSeconds` above should be updated as well

Reviewer: Also ideally the [Fields Reference](https://argo-workflows.readthedocs.io/en/latest/fields/) should be updated as well; who's source is in `cron_workflow_types.go`. Although that one isn't incorrect at least, it's just different

### Crash Recovery

> If the `workflow-controller` crashes (and hence the `CronWorkflow` controller), there are some options you can set to ensure that `CronWorkflows` that would have been scheduled while the controller was down can still run. Mainly `startingDeadlineSeconds` can be set to specify the maximum number of seconds past the last scheduled run of a `CronWorkflow` during which a missed run will still be executed.

Reviewer: ```suggestion
If the Controller crashes, you can ensure that any missed schedules still run.

With `startingDeadlineSeconds` you can specify a maximum grace period past the last scheduled time during which it will still run.
```

- simplify, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
  - `CronWorkflows` is redundant, since this the Cron Workflow page
- prefer 1 sentence per line of markdown, per [style guide](https://argo-workflows.readthedocs.io/en/release-3.5/doc-changes/)
- use the phrase "grace period", as suggested in the issue

These style guides weren't used back when this doc was originally written, but we should update it to the latest style guides whenever we change it

> For example, if a `CronWorkflow` that runs every minute is last run at 12:05:00, and the controller crashes between 12:05:55 and 12:06:05, then the expected execution time of 12:06:00 would be missed. However, if `startingDeadlineSeconds` is set to a value greater than 5 (the amount of time passing between the last scheduled run time of 12:06:00 and the current time of 12:06:05), then a single instance of the `CronWorkflow` will be executed exactly at 12:06:05.

Reviewer: ```suggestion
For example, if a `CronWorkflow` that runs every minute is last run at 12:05:00, and the controller crashes between 12:05:55 and 12:06:05, then the expected execution time of 12:06:00 would be missed.
However, if `startingDeadlineSeconds` is set to a value greater than 5 (the time passed between the last scheduled time of 12:06:00 and the current time of 12:06:05), then a single instance of the `CronWorkflow` will be executed exactly at 12:06:05.
```

same as above
